apps/game_backend/lib/game_backend/curse_of_mirra/quests.ex

```diff
@@ -274,29 +316,37 @@ defmodule GameBackend.CurseOfMirra.Quests do
   defp accumulate_objective_progress_by_scope("day", value), do: value
   defp accumulate_objective_progress_by_scope("match", _value), do: 1
 
-  def completed_quest?(%UserQuest{quest: %Quest{} = quest}, arena_match_results) do
+  def completed_quest?(nil, _arena_match_results) do
+    false
+  end
+
+  def completed_quest?(quest, arena_match_results) do
     progress =
       arena_match_results
       |> filter_results_that_meet_quest_conditions(quest.conditions)
       |> Enum.reduce(0, fn arena_match_result, acc ->
-        type = String.to_atom(quest.objective["match_tracking_field"])
+        type = String.to_existing_atom(quest.objective["match_tracking_field"] || quest.objective.match_tracking_field)
```

Comment:
Are you certain these atoms exist everytime at this point? Otherwise we could do a wiser approach by creating the atoms explicitly. For example: `def type_atom("foo"), do :foo, def type_atom("bar"), do :bar, ...`

Reply:
Since they are a required field from the match_result table i think they're created yes, do you think this could crash at some point?

Reply:
I think you misunderstood my comment. I'm not talking about the field in the database/schema. I'm talking about the Atom you're trying to cast to.

Let's say objective["field"] has the value "value". If you do String.to_existing_atom("value") and :value doesn't exists, it will crash.
I believe is way better to explicit the atom result so we:
1. Use it if it exists.
2. Define it in a controlled scope (only the ones we define will be created).

Reply:
Thanks for explaining! 
what you meant was to create these atoms in a controlled environment where we can define a limit and make sure they won't crash in any unexpected scenario, correct me if i'm wrong